app/models/katello/content_view.rb

```diff
@@ -16,14 +16,11 @@ class ContentView < Katello::Model
     has_many :content_view_versions, :class_name => "Katello::ContentViewVersion", :dependent => :destroy
     alias_method :versions, :content_view_versions
 
-    has_many :content_view_components, :class_name => "Katello::ContentViewComponent", :dependent => :destroy
-    has_many :components, :through => :content_view_components, :class_name => "Katello::ContentViewVersion",
-                          :source => :content_view_version do
-      def <<(*_args)
-        # this doesn't go through validation and generate a nice error message
-        fail "Adding components without doing validation is not supported"
-      end
-    end
+    has_many :content_view_components, :class_name => "Katello::ContentViewComponent", :dependent => :destroy,
+             :inverse_of => :composite_content_view, :foreign_key => :composite_content_view_id
+
+    has_many :latest_content_view_components, :source => :content_view_components, :class_name => "Katello::ContentViewComponent",
+             :dependent => :destroy, :inverse_of => :content_view, :foreign_key => :content_view_id
```

Comment:
So I need to handle the case where you either remove the CV or the cv component (for the latest cv cases ).  I had this association to tackle that, because I couldn't figure out how else to remove those.